staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -275,11 +274,6 @@ func DefaultKubernetesUserAgent() string {
 // running inside a pod running on kubernetes. It will return an error if
 // called from a process not running in a kubernetes environment.
 func InClusterConfig() (*Config, error) {
-	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
```

Comment:
Oh, sorry, I misread what you wrote. Nevermind on that front.

Do you foresee an issue with what @mfojtik has done here - trying dns first, then falling back to env var for backwards compatibility?